saleor/order/calculations.py

```diff
@@ -57,16 +63,27 @@ def fetch_order_prices_if_expired(
     if order.status not in ORDER_EDITABLE_STATUS:
         return order, lines
 
-    if not force_update and not order.should_refresh_prices:
+    expired_line_ids = get_expired_line_ids(order, lines)
+    if not force_update and not order.should_refresh_prices and not expired_line_ids:
```

Comment:
As right now you need to match two conditions to exit from price recalculation, maybe it would be good for the readability to separate them to two variables? Something like this:

```suggestion
    no_need_to_calculate_taxes = not force_update and not order.should_refresh_prices
    no_need_to_calculate_expired_base = not expired_line_ids
    if no_need_to_calculate_taxes and no_need_to_calculate_expired_base:
```

Right now it is easier to see, that to go inside this `if`, we need to have valid taxes, and valid base_prices.

Reply:
Also, for better readability, I would also recommend to rename `force_update` to `force_tax_update`. As this could be misleading, that `force_update` is responsible for re-calculate taxes and recalculate base prices

Reply:
What about something like:
```python
need_recalculation = (
   order.status in ORDER_EDITABLE_STATUS 
   and (force_update or order.should_refresh_prices or expired_line_ids)
   )
if not need_recalculation:
   return order, lines
```

Reply:
https://github.com/saleor/saleor/pull/17428 here is a draft PR with proposal to increase readability of whole `fetch_order_pirces_if_expired` function